zrml/prediction-markets/src/lib.rs

```diff
@@ -215,22 +220,29 @@ mod pallet {
             Self::deposit_event(Event::MarketDestroyed(market_id));
 
             // Weight correction
+            // The DestroyOrigin should not pay fees for providing this service
             if market_status == MarketStatus::Reported {
-                Ok(Some(T::WeightInfo::admin_destroy_reported_market(
-                    total_accounts.saturated_into(),
-                    share_accounts.saturated_into(),
-                    outcome_assets_amount.saturated_into(),
-                ))
-                .into())
+                Ok((
+                    Some(T::WeightInfo::admin_destroy_reported_market(
+                        total_accounts.saturated_into(),
+                        share_accounts.saturated_into(),
+                        outcome_assets_amount.saturated_into(),
+                    )),
+                    Pays::No,
+                )
+                    .into())
             } else if market_status == MarketStatus::Disputed {
-                Ok(Some(T::WeightInfo::admin_destroy_disputed_market(
-                    total_accounts.saturated_into(),
-                    share_accounts.saturated_into(),
-                    outcome_assets_amount.saturated_into(),
-                ))
-                .into())
+                Ok((
+                    Some(T::WeightInfo::admin_destroy_disputed_market(
+                        total_accounts.saturated_into(),
+                        share_accounts.saturated_into(),
+                        outcome_assets_amount.saturated_into(),
+                    )),
+                    Pays::No,
+                )
+                    .into())
             } else {
-                Ok(None.into())
+                Ok((None, Pays::No).into())
```

Comment:
Early return errors still lead to fees as I understand it correctly? Because you could have used `Pays::No` in the `[pallet::weight]` annotation as default.

Reply:
:fire: Great catch :fire: Mitigated in [da863bd](https://github.com/zeitgeistpm/zeitgeist/pull/744/commits/da863bddcd6f90ac7adfed483e8865d81437f49a)